relay-event-normalization/src/normalize/mod.rs

```diff
@@ -1416,6 +1480,60 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_merge_builtin_measurement_keys() {
+        let foo = BuiltinMeasurementKey::new("foo", MetricUnit::Duration(DurationUnit::Hour));
+        let bar = BuiltinMeasurementKey::new("bar", MetricUnit::Duration(DurationUnit::Day));
+        let baz = BuiltinMeasurementKey::new("baz", MetricUnit::Duration(DurationUnit::Week));
+
+        let proj = MeasurementsConfig {
+            builtin_measurements: vec![foo.clone(), bar.clone()],
+            max_custom_measurements: 3,
+        };
+
+        let glob = MeasurementsConfig {
+            // The 'bar' here will be ignored since it's a duplicate from the project level.
+            builtin_measurements: vec![baz.clone(), bar.clone()],
+            max_custom_measurements: 3,
+        };
+        let dynamic_config = DynamicMeasurementsConfig::new(Some(&proj), Some(&glob));
+
+        let keys = dynamic_config.builtin_measurement_keys().collect_vec();
+
+        assert_eq!(keys, vec![&foo, &bar, &baz]);
+    }
+
+    #[test]
+    fn test_max_custom_measurement() {
+        // Empty configs will return a None value for max measurements.
+        let dynamic_config = DynamicMeasurementsConfig::new(None, None);
+        assert!(dynamic_config.max_custom_measurements().is_none());
+
+        let proj = MeasurementsConfig {
+            builtin_measurements: vec![],
+            max_custom_measurements: 3,
+        };
+
+        let glob = MeasurementsConfig {
+            builtin_measurements: vec![],
+            max_custom_measurements: 4,
+        };
+
+        // If just global or just project is available, return their max custom measurements value.
+        {
+            let dynamic_config = DynamicMeasurementsConfig::new(Some(&proj), None);
+
+            assert_eq!(dynamic_config.max_custom_measurements().unwrap(), &3);
+
+            let dynamic_config = DynamicMeasurementsConfig::new(None, Some(&glob));
+            assert_eq!(dynamic_config.max_custom_measurements().unwrap(), &4);
+        }
```

Comment:
Q: is there a reason to make this a block?

Reply:
its to make it more clear that the comment above it is referring to whats inside the block, is there a better way to do this?

Reply:
ok changed it to be more conventional